packages/lab/src/dialog/DialogTitle.tsx

```diff
@@ -13,36 +14,50 @@ import dialogTitleCss from "./DialogTitle.css";
 
 const withBaseName = makePrefixer("saltDialogTitle");
 
-interface DialogTitleProps extends ComponentPropsWithoutRef<"h2"> {
+interface DialogTitleProps
+  extends Omit<ComponentPropsWithoutRef<"div">, "title"> {
   /**
    * The status of the Dialog
-   * */
+   */
   status?: ValidationStatus | undefined;
   /**
-   * Displays the accent bar in the Dialog Title
-   * */
+   * Displays the accent bar in the Dialog Title */
   disableAccent?: boolean;
+  /**
+   * Displays the Dialog Title in a H2 component
+   */
+  title: ReactNode;
+  /**
+   * Displays the Dialog Subtitle in a Label component
+   **/
+  subtitle?: ReactNode;
+
+  className?: string;
 }
 
 export const DialogTitle = ({
-  children,
   className,
+  title,
+  subtitle,
   disableAccent,
   status: statusProp,
   ...rest
 }: DialogTitleProps) => {
-  const { status: statusContext } = useDialogContext();
+  const { status: statusContext, id } = useDialogContext();
   const targetWindow = useWindow();
   useComponentCssInjection({
     testId: "salt-dialog-title",
     css: dialogTitleCss,
     window: targetWindow,
   });
 
+  const subtitleId = `${id}-subtitle`;
+  const titleId = `${id}-title`;
```

Comment:
subtitle and title ids passed down to their respective components, enables screen reader to announce the respective content from aria-labelelledby in `Dialog` root

Reply:
Have you tried just applying the id to the root of the component?